core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -229,15 +296,41 @@ public void start(Boolean isPrimary) throws IOException {
           ServerConfiguration.global(), mMasterContext.getUserState()));
       getExecutorService().submit(
           new HeartbeatThread(HeartbeatContext.MASTER_LOG_CONFIG_REPORT_SCHEDULING,
-          new LogConfigReportHeartbeatExecutor(),
-          (int) ServerConfiguration.getMs(PropertyKey.MASTER_LOG_CONFIG_REPORT_HEARTBEAT_INTERVAL),
+              new LogConfigReportHeartbeatExecutor(),
+              (int) ServerConfiguration
+                  .getMs(PropertyKey.MASTER_LOG_CONFIG_REPORT_HEARTBEAT_INTERVAL),
               ServerConfiguration.global(), mMasterContext.getUserState()));
 
       if (ServerConfiguration.getBoolean(PropertyKey.MASTER_DAILY_BACKUP_ENABLED)) {
         mDailyBackup = new DailyMetadataBackup(this, Executors.newSingleThreadScheduledExecutor(
             ThreadFactoryUtils.build("DailyMetadataBackup-%d", true)), mUfs);
         mDailyBackup.start();
       }
+      if (mState.getClusterID() == INVALID_CLUSTER_ID) {
+        try (JournalContext context = createJournalContext()) {
+          String clusterID = java.util.UUID.randomUUID().toString();
+          mState.applyAndJournal(context, clusterID);
+          LOG.info("Created new cluster ID {}", clusterID);
+        }
+        if (ServerConfiguration.getBoolean(PropertyKey.MASTER_UPDATE_CHECK_ENABLED)) {
+          try {
+            String latestVersion = UpdateCheck.getLatestVersion(mState.getClusterID(),
+                FormatUtils.parseTimeSize("60sec"),
```

Comment:
why are we using `FormatUtils.parseTimeSize`. Can't we just have a constant long for the time? Also, all these timeouts are a minute. Are they too long? Would this mean startup could take up to 3 additional minutes in the worst case?

Reply:
reduced each of the timeouts to 3 secs